testplan/common/utils/doc.py

```diff
@@ -0,0 +1,64 @@
+"""This module contains document related utilites."""
+
+import os
+import re
+import six
+
+from testplan.common.config import Config, ConfigOption
+from testplan.common.entity import Entity
+from testplan.common.utils import logger
+
+
+def gendoc(cls):
```

Comment:
This function is clever but unfortunately it's a bit too clever. We need docstrings to be defined statically so that a) they can be read by us when reading the code as reference and b) they can be understood by IDE highlighters and the builtin help() function.